torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -283,6 +289,198 @@ inline void errorIfCapturingNonCapturableNCCL(c10::cuda::CaptureStatus status) {
 
 } // namespace
 
+namespace {
+std::string write_pickle(const c10::IValue& v) {
```

Comment:
nit: maybe name 'pickle_str' or something? people might expect this to create a file